integration-tests/suites/http_endpoint_availability.go

```diff
@@ -0,0 +1,47 @@
+package suites
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/stackrox/collector/integration-tests/pkg/collector"
+)
+
+type HttpEndpointAvailabilityTestSuite struct {
+	IntegrationTestSuiteBase
+	Port             int
+	CollectorOptions collector.StartupOptions
+	Endpoints        []string
+}
+
+func (s *HttpEndpointAvailabilityTestSuite) SetupSuite() {
+	s.StartCollector(false, &s.CollectorOptions)
+}
+
+func (s *HttpEndpointAvailabilityTestSuite) TestAvailability() {
+	for _, endpoint := range s.Endpoints {
+		url := fmt.Sprintf("http://localhost:%d%s", s.Port, endpoint)
+
+		fmt.Println(url)
+
+		resp, err := http.Get(url)
+		s.Require().NoError(err)
+		s.Require().NotNil(resp)
+		if resp != nil {
+			s.Require().Equal(resp.StatusCode, http.StatusOK, "HTTP status code")
+
+			var buf bytes.Buffer
+			defer resp.Body.Close()
+			_, err = buf.ReadFrom(resp.Body)
+			s.Require().NoError(err)
+			s.Require().True(json.Valid(buf.Bytes()))
```

Comment:
Any way we could do some validation on the actual content of the JSON response? With this test the endpoint could be returning an empty object incorrectly and this wouldn't catch it.

Reply:
The test results from a request of @erthalion, and is very minimally testing that we don't break the introspection API without noticing it. Testing the content is more work and is covered by the MockSensor integration tests (not through the introspection API though, but this is not a customer feature).